workflows/multiplesequencealign.nf

```diff
@@ -0,0 +1,248 @@
+include { ALIGN                  } from '../subworkflows/local/align'
+include { EVALUATE               } from '../subworkflows/local/evaluate'
+include { TEMPLATES              } from '../subworkflows/local/templates'
+
+//
+// MODULE: local modules
+//
+include { PREPARE_MULTIQC } from '../modules/local/prepare_multiqc'
+include { PREPARE_SHINY   } from '../modules/local/prepare_shiny'
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    IMPORT NF-CORE MODULES/SUBWORKFLOWS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+//
+// MODULE: Installed directly from nf-core/modules
+//
+include { UNTAR                          } from '../modules/nf-core/untar/main'
+include { CSVTK_JOIN as MERGE_STATS_EVAL } from '../modules/nf-core/csvtk/join/main.nf'
+include { PIGZ_COMPRESS                  } from '../modules/nf-core/pigz/compress/main'
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    RUN MAIN WORKFLOW
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+workflow MULTIPLESEQUENCEALIGN {
+
+    take:
+    ch_input    // channel: [ meta, path(sequence.fasta), path(reference.fasta), path(dependency_files.tar.gz), path(templates.txt) ]
+    ch_tools    // channel: [ val(guide_tree_tool), val(args_guide_tree_tool), val(alignment_tool), val(args_alignment_tool) ]
+    ch_versions // channel: [ path(versions.yml) ]
+
+    main:
+    ch_multiqc_files             = Channel.empty()
+    ch_multiqc_table             = Channel.empty()
+    evaluation_summary           = Channel.empty()
+    stats_summary                = Channel.empty()
+    stats_and_evaluation_summary = Channel.empty()
+    ch_shiny_stats               = Channel.empty()
+
+    ch_input
+        .map {
+            meta, fasta, ref, str, template ->
+                [ meta, file(fasta) ]
+        }
+        .set { ch_seqs }
+
+    ch_input
+        .filter { it[2].size() > 0}
+        .map {
+            meta, fasta, ref, str, template ->
+                [ meta, file(ref) ]
+        }
+        .set { ch_refs }
+
+    ch_input
+        .filter { it[4].size() > 0}
+        .map {
+            meta, fasta, ref, str, template ->
+                [ meta, file(template) ]
+        }
+        .set { ch_templates }
+
+    ch_input
+        .map {
+            meta, fasta, ref, str, template ->
+                [ meta, str ]
+        }
+        .filter { it[1].size() > 0 }
+        .set { ch_dependencies }
+
+    // ----------------
+    // DEPENDENCY FILES
+    // ----------------
+    // Dependency files are taken from a directory.
+    // If the directory is compressed, it is uncompressed first.
+    ch_dependencies
+        .branch {
+            compressed:   it[1].endsWith('.tar.gz')
+            uncompressed: true
+        }
+        .set { ch_dependencies }
+
+    UNTAR (ch_dependencies.compressed)
```

Comment:
Versions are not collected for this module

Reply:
done